File: CleanCodeJava/app/src/main/java/io/chuumong/cleancodejava/di/AppComponent.java
package io.chuumong.cleancodejava.di;

import javax.inject.Singleton;

import dagger.Component;
import io.chuumong.cleancodejava.di.main.MainComponent;
import io.chuumong.cleancodejava.di.main.MainModule;

/**
 * Created by JongHunLee on 2017-02-22.
 */


@Singleton
@Component(modules = {AppModule.class, ApiModule.class})
public interface AppComponent {

    MainComponent plus(MainModule module);
}

File: CleanCodeKotlin/app/src/main/java/io/chuumong/cleancodekotlin/di/AppComponent.kt
package io.chuumong.cleancodekotlin.di

import dagger.Component
import io.chuumong.cleancodekotlin.di.main.MainComponent
import io.chuumong.cleancodekotlin.di.main.MainModule
import java.util.*
import javax.inject.Singleton

/**
 * Created by JongHunLee on 2017-02-23.
 */

@Singleton
@Component(modules = arrayOf(AppModule::class, ApiModule::class))
interface AppComponent {
    fun plus(module: MainModule): MainComponent
}